File: packages/android/src/main/java/com/kozmos/components/Alert/Alert.figma.kt
package com.kozmos.components.alert

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=83-308")
class KozmosAlertConnect {
    @FigmaProperty(FigmaType.Enum, "Variant")
    val status: AlertStatus = Figma.mapping(
        "Default" to AlertStatus.Default,
        "Destructive" to AlertStatus.Error,
        "Success" to AlertStatus.Success,
        "Warning" to AlertStatus.Warning,
        "Info" to AlertStatus.Info
    )

    @FigmaProperty(FigmaType.Text, "Title")
    val title: String = "Heads up"

    @FigmaProperty(FigmaType.Text, "Description")
    val description: String = "Use alert description for important status detail."

    @Composable
    fun ComponentExample() {
        KozmosAlert(status = status) {
            KozmosAlertIcon(status = status)
            KozmosAlertContent {
                KozmosAlertTitle(title, color = alertForegroundColor(status))
                KozmosAlertDescription(description, color = alertForegroundColor(status))
            }
        }
    }
}

File: packages/ios/Sources/Components/Alert/Alert.figma.swift
import SwiftUI
import Figma

struct KozmosAlertConnect: FigmaConnect {
    let component = KozmosAlert<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=83-308"

    @FigmaEnum(
        "Variant",
        mapping: [
            "Default": KozmosAlert<AnyView>.KozmosAlertVariant.`default`,
            "Destructive": KozmosAlert<AnyView>.KozmosAlertVariant.destructive,
            "Success": KozmosAlert<AnyView>.KozmosAlertVariant.success,
            "Warning": KozmosAlert<AnyView>.KozmosAlertVariant.warning,
            "Info": KozmosAlert<AnyView>.KozmosAlertVariant.info
        ]
    )
    var alertVariant: KozmosAlert<AnyView>.KozmosAlertVariant = .default

    @FigmaString("Title")
    var title: String = "Heads up"

    @FigmaString("Description")
    var description: String = "Use alert description for important status detail."

    var body: some View {
        KozmosAlert(variant: self.alertVariant) {
            AnyView(
                HStack(alignment: .top, spacing: KozmosDimensions.primitivesLayoutSpacing150) {
                    KozmosAlertIcon(variant: self.alertVariant)
                    KozmosAlertContent {
                        KozmosAlertTitle(self.title, color: self.alertVariant.foregroundColor)
                        KozmosAlertDescription(self.description, color: self.alertVariant.foregroundColor)
                    }
                }
            )
        }
    }
}
